interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java

```diff
@@ -359,28 +359,31 @@ public void pingPong() throws Exception {
             .build());
 
     @SuppressWarnings("unchecked")
-    StreamObserver<StreamingOutputCallResponse> responseObserver = mock(StreamObserver.class);
+    StreamRecorder<StreamingOutputCallResponse> responseObserver = StreamRecorder.create();
     StreamObserver<StreamingOutputCallRequest> requestObserver
         = asyncStub.fullDuplexCall(responseObserver);
     for (int i = 0; i < requests.size(); i++) {
       requestObserver.onNext(requests.get(i));
-      verify(responseObserver, timeout(operationTimeoutMillis())).onNext(goldenResponses.get(i));
```

Comment:
But that's not all of what this was trying to check. It was making sure that the channel is full-duplex. We should be able to receive the response before sending the next request.

HTTP is commonly implemented as half-duplex, where the client's request is fully received before the server starts processing. Being able to do ping-pong within a single stream is a very, very important case that can break in-practice.